src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py

```diff
@@ -11,13 +11,216 @@
 """
 
 import os
+from typing import Optional
 
 from zowe.core_for_zowe_sdk import SdkApi
 from zowe.core_for_zowe_sdk.exceptions import FileNotFound
 from zowe.zos_files_for_zowe_sdk.constants import FileType, zos_file_constants
 
 _ZOWE_FILES_DEFAULT_ENCODING = zos_file_constants["ZoweFilesDefaultEncoding"]
 
+
+class DatasetOption:
+
+    def __init__(
+        self,
+        like: str = None,
+        volser: Optional[str] = None,
+        unit: Optional[str] = None,
+        dsorg: Optional[str] = None,
+        alcunit: Optional[str] = None,
+        primary: Optional[int] = None,
+        secondary: Optional[int] = None,
+        dirblk: Optional[int] = None,
+        avgblk: Optional[int] = None,
+        recfm: Optional[str] = None,
+        blksize: Optional[int] = None,
+        lrecl: Optional[int] = None,
+        storclass: Optional[str] = None,
+        mgntclass: Optional[str] = None,
+        dataclass: Optional[str] = None,
+        dsntype: Optional[str] = None,
+    ) -> None:
+        self.__like = like
+        self.__volser = volser
+        self.__unit = unit
+        self.dsorg = dsorg
+        self.alcunit = alcunit
+        self.primary = primary
+        self.secondary = secondary
+        self.__dirblk = dirblk
+        self.__avgblk = avgblk
+        self.recfm = recfm
+        self.blksize = blksize
+        self.__lrecl = lrecl
+        self.__storclass = storclass
+        self.__mgntclass = mgntclass
+        self.__dataclass = dataclass
+        self.__dsntype = dsntype
```

Comment:
Should we define all the properties without using double underscores to ensure they get validated (`self.volser = volser`, `self.unit = unit`, etc)? With the exception of `like`, I believe we discussed it's a special property and we wouldn't allow users to set it outside of the constructor.